ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/WatchedImages/WatchedImagesModal.tsx

```diff
@@ -21,53 +33,121 @@ function WatchedImagesModal({
-        (name: string) => watchImage(name),
-        { onSuccess: onWatchedImagesChange }
-    );
+    const watchedImagesFn = useCallback(getWatchedImages, []);
+    const currentWatchedImagesRequest = useRestQuery(watchedImagesFn);
+
+    const watchImageMutation = useRestMutation((name: string) => watchImage(name), {
+        onSuccess: onWatchedImagesChange,
+    });
+
+    const unwatchImageMutation = useRestMutation((name: string) => unwatchImage(name), {
+        onSuccess: () => {
+            onWatchedImagesChange();
+            currentWatchedImagesRequest.refetch();
+        },
+    });
 
     function onCloseModal() {
         onClose();
-        reset();
+        watchImageMutation.reset();
+        unwatchImageMutation.reset();
     }
 
+    const watchedImages = currentWatchedImagesRequest.data ?? [];
+
     return (
         <Modal
             title="Manage watched images"
             isOpen={isOpen}
             onClose={onCloseModal}
             variant="medium"
-            showClose={!isLoading}
-            onEscapePress={isLoading ? noop : onCloseModal}
+            showClose={false}
+            onEscapePress={watchImageMutation.isLoading ? noop : onCloseModal}
             actions={[
-                <Button key="Close" onClick={onCloseModal} isDisabled={isLoading}>
+                <Button
+                    key="Close"
+                    onClick={onCloseModal}
+                    isDisabled={watchImageMutation.isLoading}
+                >
                     Close
                 </Button>,
             ]}
         >
             <Flex direction={{ default: 'column' }}>
-                {isSuccess && (
+                {watchImageMutation.isSuccess && (
                     <Alert
                         variant="success"
                         isInline
                         title="The image was successfully added to the watch list"
                     >
-                        {data ?? ''}
+                        {watchImageMutation.data ?? ''}
                     </Alert>
                 )}
-                {isError && (
+                {watchImageMutation.isError && (
                     <Alert
                         variant="danger"
                         isInline
                         title="There was an error adding the image to the watch list"
                     >
-                        {getAxiosErrorMessage(error)}
+                        {getAxiosErrorMessage(watchImageMutation.error)}
                     </Alert>
                 )}
                 <WatchedImagesForm
                     defaultWatchedImageName={defaultWatchedImageName}
-                    watchImage={mutate}
+                    watchImage={watchImageMutation.mutate}
+                    watchedImagesRequest={currentWatchedImagesRequest}
                 />
+                <Divider component="div" />
+                {unwatchImageMutation.isSuccess && (
```

Comment:
@zhenpesky @mansursyed @dvail I was wondering about how we show the alerts in different locations depending on what they're alerting on. Is this what we want to do here? I personally prefer the alerts to always show up in the same location, but it was interesting to see them in separate locations depending on what they're alerting on. Thoughts?

![266722962-c41703e7-12ba-4b2d-ba7d-b0ca4cfaee3a](https://github.com/stackrox/stackrox/assets/4805485/9a5f1623-2753-48c6-9a81-b82ca1d02334)
![266723059-9472ba80-c50d-4eb9-a13d-ebb171bee121](https://github.com/stackrox/stackrox/assets/4805485/396ec560-7bab-449e-834d-513e243502f6)

Reply:
As decided at the UX sync, I've pushed an update that moves all alerts in this modal to the top.